code/modules/research/xenobiology/vatgrowing/microscope.dm

```diff
@@ -5,28 +5,37 @@
 	icon_state = "microscope"
 	var/obj/item/petri_dish/current_dish
 
-/obj/structure/microscope/attacked_by(obj/item/I, mob/living/user)
-	if(!istype(I, /obj/item/petri_dish))
-		return ..()
-	if(current_dish)
-		to_chat(user, span_warning("There is already a petridish in \the [src]."))
-		return
-	to_chat(user, span_notice("You put [I] into \the [src]."))
-	current_dish = I
-	current_dish.forceMove(src)
+/obj/structure/microscope/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(istype(tool, /obj/item/petri_dish))
+		if(current_dish)
+			balloon_alert(user, "already has a dish!")
+			return ITEM_INTERACT_FAILURE
+		current_dish = tool
+		current_dish.forceMove(src)
```

Comment:
Also, this really should be using `transferItemToLoc` (`if(!user.transferItemToLoc(tool, src) return`) to respect nodrop. You don't have to change it since it was like that before but it would be appreciated.